metals/src/main/scala/scala/meta/internal/metals/Indexer.scala

```diff
@@ -39,6 +39,9 @@ import scala.meta.io.AbsolutePath
 import ch.epfl.scala.{bsp4j => b}
 import org.eclipse.lsp4j.Position
 
+// todo https://github.com/scalameta/metals/issues/4788
+// clean () =>, use plain values
```

Comment:
This one should be fairly simple right? Are you sure you don't want to just tackle this right away? Or do you still want to merge and then send in a follow-up for this? The reason I ask is because you already dit this for `tables` in here. So its odd that you did one but not the others.

Reply:
One should be really careful when replacing `() =>`, in most cases this will be straighforward change, but sometimes `() =>` was used to capture mutable state. I already made that mistake and it took some time to find culprit. 

> The reason I ask is because you already dit this for tables in here. So its odd that you did one but not the others.

Well, doing it for tables made me realize how much such stuff lives in codebase. I saw those 2 classes, but probably there are more. So if I did it for Tables, Indexer then why not for others? ;) This PR is already big and I strongly believe in small (if possible), with well-defined scope, short-living ones. That's why I'm pushing all of those things into smaller, auxiliary tasks. They're relatively easy so maybe even someone else will do them, who knows :D